### ISSUE TITLE
getStaticProps || getInitialProps

### ISSUE THREAD

**Issue by danielweil** [ShanonJackson/nextjs-azure-function#1]
Do you plan to implement these features? If not, how should I proceed?

Thanks!

**Comment by skprasadu** [comment]
> Do you plan to implement these features? If not, how should I proceed?
> 
> Thanks!

Any plans for this, I also need this.

**Comment by ShanonJackson** [comment]
Hi guys and @skprasadu 

Since this library was released around 10-20 days ago we finally upgraded our NextJS version past "12.0.3" which should be the last version that's compatible with this library, for newer versions we have a methodology for deployment that increases speed significantly (build times faster by 2x and endpoints faster).

This new methodology leverages the "output: 'standalone'" method that was introduced in newer versions of NextJS which means this library would no longer have to rely on so many internal libraries to actually get the desired output.

**in summary:**
Yes, I'm willing to support it. But only for newer versions of NextJS >=12.0.3 with a preference for the absolute latest (12.2.3), I can commit the upgrade this weekend and then commit support for "getInitialProps" potentially the following weekend.

In meantime happy to help anyone trying anyone trying to host NextJS apps on Azure with advice as we've had amazing success hosting our projects there over the last 2 years.

**Comment by ShanonJackson** [comment]
@danielweil @skprasadu 

https://github.com/ShanonJackson/nextjs-azure

It's fairly far along but still work to be done. I'd recommend using our infra more than happy to help in any way documentation website is en-route

Actually need some people to help me test it on their projects as I don't use any of the "getX" methods in any of my projects.


Our infrastructure setup is roughly 30-50% faster than Vercel, has better caching policies, SSL offloading @Edge and various other improvements. This is just the start of the performance journey as when I have the time I have a list of around 10 more optimizations.

**Comment by rbourdon** [comment]
> @danielweil @skprasadu
> 
> https://github.com/ShanonJackson/nextjs-azure
> 
> It's fairly far along but still work to be done. I'd recommend using our infra more than happy to help in any way documentation website is en-route
> 
> Actually need some people to help me test it on their projects as I don't use any of the "getX" methods in any of my projects.
> 
> Our infrastructure setup is roughly 30-50% faster than Vercel, has better caching policies, SSL offloading @edge and various other improvements. This is just the start of the performance journey as when I have the time I have a list of around 10 more optimizations.

What became of this project?
